docs/examples/use_cases/tensorflow/yolov4/src/dali/ops.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2021 Kacper Kluk, Piotr Kowalewski. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import nvidia.dali as dali
+
+def input(file_root, annotations_file, shard_id, num_threads, device):
+    inputs, bboxes, classes = dali.fn.readers.coco(
+        file_root=file_root,
+        annotations_file=annotations_file,
+        ltrb=True,
+        shard_id=shard_id,
+        num_shards=num_threads,
+        ratio=True,
+        random_shuffle=True
+    )
+    images = dali.fn.decoders.image(inputs, device=device, output_type=dali.types.RGB)
+
+    return images, bboxes, classes
+
+# Converts ltrb bbox coordinates to xywh, where xy denotes coordinates of a bbox center.
+def ltrb_to_xywh(bboxes):
+    return dali.fn.coord_transform(
+        bboxes,
+        M=[0.5,  0.0,  0.5,  0.0,
+           0.0,  0.5,  0.0,  0.5,
+          -1.0,  0.0,  1.0,  0.0,
+           0.0, -1.0,  0.0,  1.0]
+    )
+
+# Transforms bbox ltrb coordinates, so that they fit within a window
+#   anchored at (pos_x, pos_y) and with a shape (shape_x, shape_y).
+def bbox_adjust_ltrb(bboxes, shape_x, shape_y, pos_x, pos_y):
+    sx, sy, ex, ey = pos_x, pos_y, shape_x + pos_x, shape_y + pos_y
+    MT = dali.fn.transforms.crop(
+        to_start=dali.fn.stack(sx, sy, sx, sy),
+        to_end=dali.fn.stack(ex, ey, ex, ey)
+    )
+    return dali.fn.coord_transform(bboxes, MT=MT)
+
+# Selects if_true or if_false tensor based on the predicate value.
+# predicate should take integer value equal to either 0 or 1.
+# if_true and if_false should be 2D tensors with equal size along axis 1.
+def select(predicate, if_true, if_false):
```

Comment:
This is an ugly workaround. I think we should just write a comment that this _is_ a workaround - not to distort our user's view of how DALI should be used.

Reply:
added a comment